packages/next/client/link.tsx

```diff
@@ -204,8 +205,11 @@ class Link extends Component<LinkProps> {
   prefetch() {
     if (!this.p || typeof window === 'undefined') return
     // Prefetch the JSON page if asked (only in the client)
-    const href = this.getHref()
+    const { href, as } = this.getHrefs()
     Router.prefetch(href)
+    // Cast to any because this is a private method, might be public in the
+    // future
+    ;(Router as any).prefetchAs(as)
```

Comment:
This needs to be exposed on the router first in `client/router` under `coreMethodFields` right?

Reply:
We don't want this API to be public.